docs/writing-tests/reftest-tutorial.md

```diff
@@ -0,0 +1,262 @@
+
+    $ git checkout -b reftest-for-bdo FETCH_HEAD
+
+Now you're ready to create your patch.
+
+## Writing the test file
+
+First, we'll create a file that demonstrates the "feature under test." That is:
+we'll write an HTML document that displays some text using a `<bdo>` element.
+
+WPT has thousands of tests, so it can be daunting to decide where to put a new
+one. Generally speaking, [test files should be placed in directories
+corresponding to the specification text they are
+verifying](../test-suite-design). `<bdo>` is defined in [the "text-level
+semantics" chapter of the HTML
+specification](https://html.spec.whatwg.org/multipage/text-level-semantics.html),
+so we'll want to create our new test in the directory
+`html/semantics/text-level-semantics/the-bdo-element/`. Create a file named
+`rtl.html` and open it in your text editor.
+
+Here's one way to demonstrate the feature:
+
+```html
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>BDO element dir=rtl</title>
+<link rel="author" title="Sam Smith" href="mailto:sam@example.com">
+<link rel="help" href="https://html.spec.whatwg.org/#the-bdo-element">
+<meta name="assert" content="BDO element's DIR content attribute renders corrently given value of 'rtl'.">
+
+<p>Test passes if WAS is displayed below.</p>
+<bdo dir="rtl">SAW</bdo>
+```
+
+That's pretty dense! Let's break it down:
+
+- ```html
+  <!DOCTYPE html>
+  <meta charset="utf-8">
+  ```
+
+  We explicitly set the DOCTYPE and character set to be sure that browsers
+  don't infer them to be something we aren't expecting. We're omitting the
+  `<html>` and `<head>` tags. That's a common practice in WPT, preferred
+  because it makes tests more concise.
+
+- ```html
+  <title>BDO element dir=rtl</title>
+  ```
+  The document's title should succinctly describe the feature under test.
+
+- ```html
+  <link rel="author" title="Sam Smith" href="mailto:sam@example.com">
+  ```
+
+  You should replace the value of the `title` and `href` attributes with your
+  name and e-mail, respectively. This can help others learn who to contact with
+  questions about this test. The git history will do the
+  same thing, so you can leave this out if you prefer.
+
+- ```html
+  <link rel="help" href="https://html.spec.whatwg.org/#the-bdo-element">
+  ```
+
+  The "help" metadata should reference the specification under test so that
+  everyone understands the motivation. This is so helpful that [the CSS Working
+  Group requires it for CSS tests](css-metadata)! If you're writing a reftest
+  for a feature outside of CSS, feel free to omit this tag.
+
+- ```html
+  <meta name="assert" content="BDO element's DIR content attribute renders corrently given value of 'rtl'.">
+  ```
+
+  The "assert" metadata is a structured way for you to describe exactly what
+  you want your reftest to verify. For a direct test like the one we're writing
+  here, it might seem a liitle superfluous. It's much more helpful for
```

Comment:
```suggestion
  here, it might seem a little superfluous. It's much more helpful for
```

Reply:
Now you've gone too far. Spelling mistakes give writing personality!

Reply:
haha — I wasn’t sure if you were actually misspelled *leetle* but I took a shot in the dark and guessed it might be *little* instead